airflow/providers/amazon/aws/operators/ecs.py

```diff
@@ -471,39 +470,40 @@ def __init__(
             self.awslogs_region = self.region
 
         self.arn: str | None = None
+        self.started_by: str | None = None
+
         self.retry_args = quota_retry
         self.task_log_fetcher: EcsTaskLogFetcher | None = None
         self.wait_for_completion = wait_for_completion
 
-    @provide_session
-    def execute(self, context, session=None):
+    def execute(self, context):
         self.log.info(
             "Running ECS Task - Task definition: %s - on cluster %s", self.task_definition, self.cluster
         )
         self.log.info("EcsOperator overrides: %s", self.overrides)
 
         if self.reattach:
-            self._try_reattach_task(context)
+            # Generate deterministic UUID which refers to unique TaskInstanceKey
+            ti: TaskInstance = context["ti"]
+            self.started_by = generate_uuid(*map(str, ti.key.primary))
+            if self.do_xcom_push:
+                ti.xcom_push("started_by", self.started_by)
 
-        self._start_wait_check_task(context)
+            self._try_reattach_task()
 
-        self.log.info("ECS Task has been successfully executed")
+        self._start_wait_check_task()
 
-        if self.reattach:
-            # Clear the XCom value storing the ECS task ARN if the task has completed
-            # as we can't reattach it anymore
-            self._xcom_del(session, self.REATTACH_XCOM_TASK_ID_TEMPLATE.format(task_id=self.task_id))
```

Comment:
Do you not want the same behaviour for your implementation instead of leaving the keys behind? They're small but will accumulate over time.